script.js

```diff
@@ -47,13 +83,19 @@ function drawNewGrid() {
         drawNewGrid();
     }
 
-    else {
-        return buildGrid(newSize);
-    }
+    return buildGrid(newSize);
 }
 
+/**
+ * a.) can you think of why this might be an unsafe operation?
+ * b.) it would be cool to also reset your background colors on your tiles when you did this
+ */
 resetButton.onclick = () => {
-    while(container.firstChild) {
+    /**
+     * Any reason why you went with a while loop here instead of
+     * an if statement?
+     */
```

Comment:
My thought process was that the if statement would only fire once, whereas the while continued as long as the condition was true.

Reply:
You know - I learned something today.

Previously, to clear child DOM nodes, I had _always_ done this approach:

```js
if (container.firstChild) {
  container.innerHTML = '';
}
```

I guess I had never really thought to do it this way, because intuitively, I assumed that looping over every node would be an inherently slower operation. 

But StackOverflow seems to agree with your approach: https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript

A couple things of other things to highlight:

One optimization that you could make is to change `removeChild` to targe the `lastChild` instead of the first:
> **Looping to remove every lastChild:** An earlier edit to this answer used firstChild, but this is updated to use lastChild as in computer-science, in general, it's significantly faster to remove the last element of a collection than it is to remove the first element (depending on how the collection is implemented).

Another one, that looks really cool - is the `replaceChildren` API - so another potential solution could be:
```js
if (container.firstChild) {
  container.replaceChildren();
}
```
https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren

Above all... these are micro-optimizations. I wouldn't expect your code to execute any faster or slower if we made these changes. It's not likely to make a difference until we're dealing with hundreds of thousands of nodes, but thinking "performance-first" down the road is a good state of mind to get into.